src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,191 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Encapsulates a list of tasks for Duke.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int tasksLength;
```

Comment:
Since you initialize the ArrayList in both constructor, do we still need tasksLength?